src/optimisation/Optimisation.jl

```diff
@@ -1,324 +1,327 @@
 module Optimisation
 
 using ..Turing
+using ..OptimisationCore: ModeResult, MLE, MAP, OptimLogDensity, OptimizationContext,
```

Comment:
What's the reasoning for not keeping all this in the same module? After this PR we sort of want to support all optimizers but _through_ Optimiziation.jl. Hence it seems a bit redundant to have another "core" module when we actually just want to rely on Optimization.jl anyways (and so our "core" becomes just interfacing with Optimization.jl).

Does that make sense?

Reply:
The only reasoning was to not have OptimInterface.jl import from the Optimization.jl interface code, which seemed unaesthetical to me. If the OptimInterface.jl is going to go soon then I agree that there's no reason for the distinction.

Do we want to keep the distinction until we are confident that OptimInterface.jl is about to go though? Might want to make sure that we can e.g. turn every call to its exported functions into an equivalent call to `estimate_mode` behind the scenes, to avoid breaking user code.

Reply:
> If the OptimInterface.jl is going to go soon then I agree that there's no reason for the distinction.

> Do we want to keep the distinction until we are confident that OptimInterface.jl is about to go though? 

IMO no :man_shrugging: I get the motivation that it's unaesthetic, but IMO for the uninitiated it makes it a bit more difficult to get a sense of what's going on. We want the user to only see `maximum_a_posteriori` and `maximum_likelihood`, with `optimize` just being a thing we expose if they do `using Optim` for the sake of backwards compat (but how we implement this behind the scenes, is just using Optimization.jl)

Reply:
Done. The commit doing all the code moving is the first one of the latest batch, so if you want to see changes other than the big move, skip 4b3fbf5e5a230714afef6f5774eae6095af8115b in the diff.